homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,136 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    entities = [ZimiLight(device, api) for device in api.lights]
+
+    async_add_entities(entities)
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize an ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._light = light
+        self._light.subscribe(self)
+        self._state = False
+        self._brightness = None
+        if self._light.type == "dimmer":
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        else:
+            self._attr_color_mode = ColorMode.ONOFF
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, light.identifier)},
+            name=self._light.name,
+            suggested_area=self._light.room,
+            via_device=(DOMAIN, api.host + ":" + str(api.port)),
+        )
+        self.update()
+        _LOGGER.debug("Initialising %s in %s", self.name, self._light.room)
+
+    def __del__(self):
+        """Cleanup ZimiLight with removal of notification."""
+        self._light.unsubscribe(self)
+
+    @property
+    def name(self) -> str:
+        """Return the display name of this light."""
+        return self._name.strip()
```

Comment:
is this the main feature of the device? If yes set to `None` and it will just follow the device name

Reply:
The name of the device can be set by the end-user using the Zimi phone app.   Here I just use the name they set and strip any trailiing spaces (which caused me some debug issues in early development).

Reply:
But shouldn't that then be the device name? And this still being none so the entity follow the device name?

Reply:
I don't quite follow.   When I tried to remove this method (or return None) then the displayed name in the UI seems to default to the device identifiers.  

![image](https://github.com/user-attachments/assets/d14eee26-7847-4824-8aa1-816b73bec8a2)

I have now changed the method to read the name from the zcc device and that also works and would allow for future name update changes to propogate from zcc (if they were made from the zimi phone app).

See https://github.com/home-assistant/core/pull/129876/commits/9a71f0b9b7678a4b4261c2d8df8dce8c8964a206